source/common/router/rds_impl.cc

```diff
@@ -116,10 +117,23 @@ void RdsRouteConfigSubscription::onConfigUpdate(
     config_info_ = {new_hash, version_info};
     route_config_proto_ = route_config;
     stats_.config_reload_.inc();
-    ENVOY_LOG(debug, "rds: loading new configuration: config_name={} hash={}", route_config_name_,
-              new_hash);
-    for (auto* provider : route_config_providers_) {
-      provider->onConfigUpdate();
+
+    if (!route_config_proto_.has_vhds()) {
+      ENVOY_LOG(debug, "rds: loading new configuration: config_name={} hash={}", route_config_name_,
+                new_hash);
+
+      last_update_details_ = std::make_unique<RdsConfigUpdateDetails>(route_config_proto_,
+                                                                      last_updated_, config_info_);
+      for (auto* provider : route_config_providers_) {
+        provider->onConfigUpdate();
+      }
+      vhds_subscription_.release();
+    } else {
+      auto s = new VhdsSubscription(route_config_proto_, factory_context_, stat_prefix_,
+                                    route_config_providers_);
+      s->registerInitTargetWithInitManager(factory_context_.initManager());
+      vhds_subscription_.reset(s);
+      last_update_details_ = std::make_unique<VhdsConfigUpdateDetails>(*vhds_subscription_);
```

Comment:
I'd probably find it a little clearer as `config_update_info` (and `ConfigUpdateInfo` for types), but this is just, like, my opinion :) It did take me a few minutes to figure out what this was all about.

Reply:
done